client/src/app/game/modals/drag-cards-modal.tsx

```diff
@@ -0,0 +1,425 @@
+		if (draggedCard !== cardInfo.entity) return
+		if (!e.buttons) {
+			return
+		}
+		if (e.movementX) cardInfo.totalMovement += e.movementX
+		cardInfo.cardRef.current.style.transform = `translateX(${cardInfo.totalMovement}px)`
+	}
+
+	const testForTouch = (e: TouchEvent) => {
+		if (!draggedCard) return
+		const cardInfo = cardsInfo.find((card) => card.entity === draggedCard)
+		if (!cardInfo || !cardInfo.cardRef || !cardInfo.cardRef.current) return
+		if (!cardInfo.cardRef || !cardInfo.cardRef.current) return
+		if (!e.targetTouches) return
+		const result = e.targetTouches[0]
+		const boundingRect = cardInfo.cardRef.current.getBoundingClientRect()
+		const middle = (boundingRect.right + boundingRect.left) / 2
+		cardInfo.totalMovement += result.pageX - middle
+		cardInfo.cardRef.current.style.transform = `translateX(${cardInfo.totalMovement}px)`
+	}
+
+	useLayoutEffect(() => {
+		onCardPositionUpdate(false)
+	}, [])
+
+	const onMouseUp = () => {
+		if (draggedCard) onCardPositionUpdate(true)
+		setDraggedCard(null)
+	}
+
+	const onTouchEnd = (e: TouchEvent) => {
+		e.preventDefault()
+		onCardPositionUpdate(true)
+		setDraggedCard(null)
+	}
+
+	useLayoutEffect(() => {
+		window.addEventListener('mouseup', onMouseUp)
+		window.addEventListener('mousemove', testForSlide)
+		window.addEventListener('touchend', onTouchEnd)
+		window.addEventListener('touchmove', testForTouch)
+
+		return () => {
+			window.removeEventListener('mouseup', onMouseUp)
+			window.removeEventListener('mousemove', testForSlide)
+			window.removeEventListener('touchend', onTouchEnd)
+			window.removeEventListener('touchmove', testForTouch)
+		}
+	}, [draggedCard])
+
+	return (
+		<Modal
+			setOpen
+			title={modalData.name}
+			onClose={handleClose}
+			disableUserClose={true}
+		>
+			<Modal.Description>
+				{modalData.description}
+				<div className={css.draggableCardsContainer}>
+					<div className={css.retrievalBox}>
+						<div className={css.retrievalArea} ref={leftAreaRef}></div>
+						<div className={css.retrievalName}>
+							{modalData.leftAreaName}
+							{modalData.leftAreaMax && (
+								<span>
+									{' '}
+									({leftCards.length}/{modalData.leftAreaMax})
+								</span>
+							)}
+						</div>
+					</div>
+					<div className={css.deckSpacer}></div>
+					<div className={css.retrievalBox}>
+						<div className={css.retrievalArea} ref={rightAreaRef}></div>
+						<div className={css.retrievalName}>
+							{modalData.rightAreaName}{' '}
+							{modalData.rightAreaMax && (
+								<span>
+									{' '}
+									({rightCards.length}/{modalData.rightAreaMax})
+								</span>
+							)}
+						</div>
+					</div>
+					<div className={css.subContainer}>
+						{[...startingLeftCards, ...startingRightCards].map((card, i) => {
+							return (
+								<DraggableCard
+									entity={card.entity}
+									draggedCard={draggedCard}
+									setDraggedCard={setDraggedCard}
+									cardInfo={cardsInfo}
+									key={i}
```

Comment:
Just so this change doesn't get lost from our debugging
```suggestion
									key={card.entity}
```